### PR TITLE
HARP-13136: Adds forcePickable to fill technique, to force invisible …

### PR DESCRIPTION
**Pull request by FraukeF**
…polygons being pickable

Signed-off-by: Frauke Fritz <frauke.fritz@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

**Comment by FraukeF** [comment]
retest this please

File: @here/harp-datasource-protocol/lib/Theme.ts [hunk 1]
```diff
@@ -350,10 +351,17 @@ export interface StyleAttributes<Technique, Params> {
     maxZoomLevel?: number | JsonExpr;
 
     /**
-     * Optional. If `true`, no IDs will be saved for the geometry this style creates. Default is
-     * `false`.
+     * Optional. If `true` or `Pickability.transient`, no IDs will be saved for the geometry
+     * this style creates. Default is `Pickability.onlyVisible`, which allows all pickable and visible
+     * objects to be picked, Pickability.onlyVisible, will also allow invisible objects to be
```
Comment (atomicsulfate) [inline @here/harp-datasource-protocol/lib/Theme.ts:356]:
```suggestion
     * objects to be picked, Pickability.all, will also allow invisible objects to be
```

File: @here/harp-mapview/test/PickingRaycasterTest.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
 /*
- * Copyright (C) 2020-2021 HERE Europe B.V.
+ * Copyright (C) 2017-2020 HERE Europe B.V.
```
Comment (atomicsulfate) [inline @here/harp-mapview/test/PickingRaycasterTest.ts:2]: The years where correct before right? It should go from the year the file was created to the current year. That's what the script recently written by Thomas and German does anyway.

File: @here/harp-datasource-protocol/lib/TechniqueParams.ts [hunk 3]
```diff
@@ -209,9 +209,17 @@ export interface BaseTechniqueParams {
     category?: DynamicProperty<string>;
 
     /**
-     * Optional. If `true`, no IDs will be saved for the geometry this technique creates.
+     * Optional. If `true` or `Pickability.transient`, no IDs will be saved for the geometry
+     * this style creates. Default is `Pickability.onlyVisible`, which allows all pickable and
+     * visible objects to be picked, Pickability.xbxe, will also allow invisible objects to be
```
[the remark this answers is not in the document]
Comment (FraukeF) [reply @here/harp-datasource-protocol/lib/TechniqueParams.ts:214]: done

File: @here/harp-datasource-protocol/lib/Theme.ts (hunk 1 above)
Comment (FraukeF) [reply @here/harp-datasource-protocol/lib/Theme.ts:356]: done

File: @here/harp-mapview/test/PickingRaycasterTest.ts (hunk 2 above)
Comment (FraukeF) [reply @here/harp-mapview/test/PickingRaycasterTest.ts:2]: sorry, not sure what caused that, I did not cause that, I guess I run yarn pre-test .. I ll revert it

Comment (FraukeF) [reply @here/harp-mapview/test/PickingRaycasterTest.ts:2]: done

**Review by atomicsulfate — APPROVED** [review]

**Comment by FraukeF** [comment]
retest please

**Pull request merged**